### ISSUE TITLE
Proposal: Piping to case expressions

### ISSUE THREAD

**Issue by starbelly** [rrrene/credo#683]
### Precheck

* For proposing a new feature, be sure to name your issue "Proposal: ..."
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.x
* Erlang/Elixir version (`elixir -v`): 1.x
* Operating system: mac osx and linux

### What were you trying to do?

Running credo on a code base where their were instances of piping to case expressions. Example: 

```
foo
|> bar()
|> baz()
|> case do 
    ... 
|> eh
```

### Expected outcome

I would expect credo to complain about this.
 
### Actual outcome

Credo did not complain.

**Comment by OvermindDL1** [comment]
Piping into case is excessively useful to acting as a monad'y transformer though.

**Comment by starbelly** [comment]
@OvermindDL1 Why not just put the case in a function?

**Comment by OvermindDL1** [comment]
@starbelly It splits a singular piped idea into forcing you to look in multiple places, breaking train of thought.  More functions is entirely recommended much of the time, but in other cases it makes the flow more difficult to understand as well.

**Comment by starbelly** [comment]
@OvermindDL1 I'm all for keeping relevant behavior in one place, but in that case I would do: 

```
hmm = foo |> bar |> baz
res = case hmm do
...
end
eh(res)
```

**Comment by OvermindDL1** [comment]
It's a bit of a strawman, but look at it formatted inline:
```elixir
foo
|> bar
|> baz
|> case do
...
end
|> eh
```
Less packed expressions, shorter lines, easy to read and follow the flow of it, etc...

**Comment by starbelly** [comment]
🤷‍♂ If the core and/or community prefers this I will yield.

**Comment by OvermindDL1** [comment]
Oh I still think it would make a useful check, I know some standards don't prefer it, but it is definitely not one that should be enabled by default.

**Comment by rrrene** [comment]
@starbelly If you open a PR for this, I would probably merge it as an optional check. This way, people have a choice :+1:

**Comment by starbelly** [comment]
@rrrene Will do as soon as possible.

**Comment by whatyouhide** [comment]
Maybe worth adding `if`/`unless` too :)

**Comment by OvermindDL1** [comment]
Likely yeah.  I do pipe into `if`/`unless` at times as well.  ^.^;
